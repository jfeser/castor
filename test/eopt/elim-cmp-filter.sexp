((queries
  ("filter(a < param0, atuple([ascalar(0 as a), ascalar(0 as b)], cross))"))
 (transforms (hoist-filter elim-cmp-filter)))
