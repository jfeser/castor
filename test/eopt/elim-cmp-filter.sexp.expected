$3:
  select([a, b], aorderedidx(dedup(select([a], atuple([ascalar(0 as a),
                                                       ascalar(0 as b)], cross))), filter((a
                                                                    = 0.a), atuple([ascalar(0 as a),
                                                                    ascalar(0 as b)], cross)), , < param0))
  filter((a < param0), atuple([ascalar(0 as a), ascalar(0 as b)], cross))
  
